src/nix/app.cc

```diff
@@ -91,13 +95,13 @@ UnresolvedApp InstallableValue::toApp(EvalState & state)
             }, c.raw));
         }
 
-        return UnresolvedApp{App {
+        return UnresolvedApp { App {
             .context = std::move(context2),
             .program = program,
         }};
     }
 
-    else if (type == "derivation") {
```

Comment:
Since we always return from the previous if block, there is no need to use 'else if'

Reply:
I like the `else` because it makes it easier to see the 3 cases are disjoint without having to read each branch seeing if they all end in `return`.

The other changes look good to me however.

Reply:
Agreed, removing else statements makes the code less structured. I am going to restore them in the next commit.